onnxruntime/core/providers/coreml/model/model.mm

```diff
@@ -169,6 +171,60 @@ Status CreateInputFeatureProvider(const std::unordered_map<std::string, OnnxTens
   conversion_buffers_out = std::move(conversion_buffers);
   return Status::OK();
 }
+
+bool IsArrayContiguous(MLMultiArray* array) {
+  int64_t batch_stride = [array.strides[0] longLongValue];
+  const auto* shape = array.shape;
+  int64_t batch_elems = 1;
+  for (unsigned long i = 1; i < shape.count; i++) batch_elems *= [shape[i] longLongValue];
+  return batch_stride == batch_elems;
+}
+
+onnxruntime::common::Status copyMLMultiArrayBuffer(const void* mlmultiarray_buffer, void* tensor_buffer,
+                                                   MLMultiArray* array_info, const OnnxTensorInfo& tensor_info,
+                                                   bool skip_buffer_size_check = true,
+                                                   const unsigned long mlmultiarray_buffer_size = 0) {
```

Comment:
can we use `std::optional<unsigned long>` instead? and avoid the default arguments if possible?

Reply:
absolutely!